main.go

```diff
@@ -495,17 +515,42 @@ func exportArtifacts(proj xcodeproj.XcodeProj, scheme string, schemeBuildDir str
 		// Copy - export
 		{
 
-			// Parent dir (main target's build dir by the provided scheme) + current target's build dir
-			sourceDir := filepath.Join(splitSchemeDir[0], "Build", splitTargetDir[1])
-
-			source := filepath.Join(sourceDir, target.ProductReference.Path)
+			// Search for the generated build artifact in the next dirs:
+			// Parent dir (main target's build dir by the provided scheme) + current target's build dir (This is a default for a nativ iOS project)
+			// current target's build dir (If the project settings uses a custom TARGET_BUILD_DIR env)
+			// .xcodeproj's directory + current target's build dir (If the project settings uses a custom TARGET_BUILD_DIR env & the project is not in the root dir)
+			sourceDirs := []string{filepath.Join(schemeDir, targetDir), schemeDir, filepath.Join(path.Dir(proj.Path), schemeDir)}
 			destination := filepath.Join(deployDir, target.ProductReference.Path)
 
-			if err := util.CopyDir(source, destination); err != nil {
-				return nil, fmt.Errorf("failed to copy the generated app to the Deploy dir")
+			// Search for the generated build artifact
+			var exported bool
+			for _, sourceDir := range sourceDirs {
+				source := filepath.Join(sourceDir, target.ProductReference.Path)
+				log.Debugf("searching for the generated app in %s", source)
+
+				cmd := util.CopyDir(source, destination)
```

Comment:
i think it would be cheaper if we would use pathutil.IsPathExists to check if the source path exists